Extensions/Firebase/firebasetools.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Firebase Tools
+ * @author arthuro555
+ */
+
+/*    Initialization of the firebase SDK    */
```

Comment:
Can you use `registerFirstRuntimeSceneLoadedCallback` to avoid having a "gdjsWatcher"?

Reply:
I was actually tlking about the big comments for structuring but yes, except I'm unsure if it wouldn't cause problems because afaik the firebase sdk loads asynchronously so it should load as soon as the data is provided to be assured to finish loading before being iused by events.